api/ldap.go

```diff
@@ -0,0 +1,101 @@
+package api
+
+import (
+	"crypto/tls"
+	"fmt"
+	"strings"
+
+	"github.com/gin-contrib/sessions"
+	"github.com/gin-gonic/gin"
+	"github.com/go-ldap/ldap/v3"
+)
+
+func Authorize(username string, password string) error {
+
+	attributes := []string{"DN", "CN"}
+	var err error
+
+	config := context.Config()
+	dn := fmt.Sprintf("%s", config.LdapDN)
+	filter := fmt.Sprintf(config.LdapFilter, username)
+
+	// connect to ldap server
+	server := fmt.Sprintf("%s", config.LdapServer)
+	conn, err := ldap.Dial("tcp", server)
+	if err != nil {
+		return err
+	}
+	defer conn.Close()
+
+	// reconnect via tls
+	err = conn.StartTLS(&tls.Config{InsecureSkipVerify: true})
```

Comment:
The choice to use "insecure" should probably be configurable.

Reply:
agreed